app/src/main/java/com/babylon/wallet/android/presentation/account/createaccount/CreateAccountViewModel.kt

```diff
@@ -106,12 +107,13 @@ class CreateAccountViewModel @Inject constructor(
 
     override fun initialState(): CreateAccountState = CreateAccountState(
         firstTime = args.requestSource?.isFirstTime() == true,
-        isCancelable = true
+        isCancelable = true,
     )
 
     fun onAccountNameChange(accountName: String) {
-        savedStateHandle[ACCOUNT_NAME] = accountName.take(ACCOUNT_NAME_MAX_LENGTH)
-        savedStateHandle[CREATE_ACCOUNT_BUTTON_ENABLED] = accountName.trim().isNotEmpty()
+        savedStateHandle[ACCOUNT_NAME] = accountName // .take(ACCOUNT_NAME_MAX_LENGTH)
+        savedStateHandle[IS_ACCOUNT_NAME_LENGTH_MORE_THAN_THE_MAX] = accountName.count() > ACCOUNT_NAME_MAX_LENGTH
+        savedStateHandle[CREATE_ACCOUNT_BUTTON_ENABLED] = accountName.trim().isNotEmpty() && accountName.count() <= ACCOUNT_NAME_MAX_LENGTH
```

Comment:
> This is the only place in the app were we replicate the state to `savedStateHandle`. Should we be doing this @jakub-rdx ?

I might be wrong but if we don't do this we cannot recreate the state properly upon app reset (when you hit Kill the app when in background) can we ?

Reply:
> Yes I know that, but why are we certainly doing this only for this view model?

Dunno, I think we treated it as low priority thing since we were not sure if the app even restore screen hierarchy/ navigation upon restart not to mention restoring state on the view.

Not sure what's the best action now, but probably to create a ticket to go through **every** view model and fix that?

Reply:
actually I tested locally and indeed it works.
I opened the create account screen. Typing something there.
Put the app in the background.
Kill process from Android Studio.
Launch the app and returns back to create account screen with whatever was typed in the text field.

Reply:
> actually I tested locally and indeed it works.
> I opened the create account screen. Typing something there.
> Put the app in the background.
> Kill process from Android Studio.
> Launch the app and returns back to create account screen with whatever was typed in the text field.

Exactly 🎉 That is how it should behave ideally 💯